File: toolchains/csharp/BloomingDirectedGraphFilterAgents.cs
namespace LangGraph;

/// <summary>Blooming-directed-graph filter agents collection (LangGraph namespace): ruby, typescript, python, csharp, java, bash, zsh, git, github, hpcc.</summary>
public static class BloomingDirectedGraphFilterAgents
{
    public static readonly IReadOnlyList<string> AgentIds = new[]
    {
        "ruby", "typescript", "python", "csharp", "java", "bash", "zsh", "git", "github", "hpcc"
    };

    public static bool IsFilterAgent(string id) => AgentIds.Contains(id);
}

File: toolchains/java/src/main/java/langgraph/BloomingDirectedGraphFilterAgents.java
package langgraph;

import java.util.*;

/**
 * Blooming-directed-graph filter agents collection (LangGraph namespace: graph filter agents).
 * Ten agents: ruby, typescript, python, csharp, java, bash, zsh, git, github, hpcc.
 */
public class BloomingDirectedGraphFilterAgents {
    public static final List<String> AGENT_IDS = Collections.unmodifiableList(Arrays.asList(
        "ruby", "typescript", "python", "csharp", "java", "bash", "zsh", "git", "github", "hpcc"
    ));

    public static List<String> agentIds() {
        return AGENT_IDS;
    }

    public static boolean isFilterAgent(String id) {
        return AGENT_IDS.contains(id);
    }
}
